-1

Tested at: 2755de2358f679d022e1c02db8a4c0b812876886


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4fa79e5f0bc0baa954914eda20a9a163cbd95d6d
https://github.com/cms-sw/cmssw/commit/0e08eaec20efd60a0776984bc93084617e380a2a
https://github.com/cms-sw/cmssw/commit/7ed5e210a34cc0896590a622bd75aabebdd9b03d
https://github.com/cms-sw/cmssw/commit/be34a0d30fe1d87a53013f71e5faa9a1327715ac
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19860/21671/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19860/21671/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19860/21671/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/DQMOffline/Trigger/plugins/TriggerDQMBase.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/DQMOffline/Trigger/plugins/BPHMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/DQMOffline/Trigger/plugins/HLTDQMObjSelector.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/DQMOffline/Trigger/plugins/JetMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/DQMOffline/Trigger/plugins/JetDQM.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/IOPool/TFileAdaptor/src/TStorageFactoryFile.cc:420:31: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
      Error("ReadBuffersSync",ostr.str().c_str());
                              ^~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-21-1100/src/IOPool/TFileAdaptor/src/TStorageFactoryFile.cc:420:31: note: treat the string as an argument to avoid this
      Error("ReadBuffersSync",ostr.str().c_str());
                              ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4fa79e5f0bc0baa954914eda20a9a163cbd95d6d
https://github.com/cms-sw/cmssw/commit/0e08eaec20efd60a0776984bc93084617e380a2a
https://github.com/cms-sw/cmssw/commit/7ed5e210a34cc0896590a622bd75aabebdd9b03d
https://github.com/cms-sw/cmssw/commit/be34a0d30fe1d87a53013f71e5faa9a1327715ac
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19860/21671/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19860/21671/git-merge-result

--- a/IOPool/TFileAdaptor/src/TStorageFactoryFile.cc
+++ b/IOPool/TFileAdaptor/src/TStorageFactoryFile.cc
@@ -414,6 +414,7 @@ TStorageFactoryFile::ReadBuffersSync(char *buf, Long64_t *pos, Int_t *len, Int_t
     std::vector<IOPosBuffer> &iov = repacker.iov();
     IOSize result = storage_->readv(&iov[0], iov.size());
     if (result != io_buffer_used) {
+      Error("ReadBuffersSync","Storage::readv returned different size result=%ld expected=%ld",result,io_buffer_used);
       return kTRUE;
     }
     xstats.tick(io_buffer_used);
@@ -477,7 +478,13 @@ TStorageFactoryFile::ReadBuffers(char *buf, Long64_t *pos, Int_t *len, Int_t nbu
   astats.tick(total);
 
   // If it didn't suceeed, pass down to the base class.
-  return success ? kFALSE : TFile::ReadBuffers(buf, pos, len, nbuf);
+  if(not success) {
+    if(TFile::ReadBuffers(buf, pos, len, nbuf)) {
+      Error("ReadBuffers", "call to TFile::ReadBuffers failed after prefetch already failed.");
+      return kTRUE;
+    }
+  }
+  return kFALSE;
 }
 
 Bool_t

